### PR TITLE
nixos/mastodon: add option changeMaxTootChars

### PR DESCRIPTION
**Pull request by Izorkin**
###### Motivation for this change
Added patch and option `changeMaxTootChars` to enable change maximum allowed character count.
Based on this patch - https://code.horrific.dev/horrific/mastodon/commit/c69bd7789c70c3f3435f5a475bf727a00cb9dad7

cc @erictapen

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by erictapen** [comment]
I'd rather not have this in Nixpkgs, sorry. Even if this wouldn't include a patch file, it would add a feature way beyond the scope of this module.

It is already possible to get what you want with this configuration on the user side (untested):

```nix
{ config, lib, ... }:{
  nixpkgs.overlays = [
    (
      self: super: {
         mastodon = super.mastodon.override {
           srcOverride = lib.fetchgit { ... }; # your Mastodon fork
         };
      }
    )
  ];
  systemd.services.mastodon-web.environment.MAX_TOOT_CHARS = "1312";
}
```

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 116104` at 9d89bdb3 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li><a href="https://gist.github.com/c46072b17501d99cb69bf1957e38d99e">mastodon</a></li></li>
  </ul>
</details>

File: nixos/modules/services/web-apps/mastodon.nix [hunk 1]
```diff
@@ -216,6 +217,12 @@ in {
         default = true;
       };
 
+      changeMaxTootChars = lib.mkOption {
+        description = "Change maximum allowed character count. Default 500 character count.";
+        type = lib.types.nullOr lib.types.int;
+        default = null;
```
Comment (SuperSandro2000) [inline nixos/modules/services/web-apps/mastodon.nix:223]: This is a bit confusing why the default is null but the text tells you 500.

Comment (Izorkin) [reply nixos/modules/services/web-apps/mastodon.nix:223]: The `MAX_TOOT_CHARS` parameter is responsible for the number of characters. If it is not in the variables, then the `MAX_CHARS` value will be by default - 500 characters.

**Comment by Izorkin** [comment]
@erictapen by default, PR does not change the default value, but only adds the ability to change the character limit.

I think that many users would like to have this opportunity. This PR makes it easy enough.

**Comment by erictapen** [comment]
@Izorkin It adds a non-trivial patch that needs to be maintained across Mastodon releases. Just following upstream releases and matching their feature set is a huge task in itself, which isn't even finished for v3.3.0 atm.

Also what comes next? Allowing this feature in would open the door to lots of other custom changes. And there are a lot of controversial contributions in the Mastodon community, that didn't make it into upstream.

I'm strongly against supporting any features that aren't supported by upstream. This also seems to me like the community consensus for most of the modules in `nixos/`.

**Comment by Izorkin** [comment]
cc @Gargron is it possible to add this patch - https://code.horrific.dev/horrific/mastodon/commit/c69bd7789c70c3f3435f5a475bf727a00cb9dad7 to the upstream? I checked it on my test machine - it works fine. By default, the symbol limit is not changed, but users are given the opportunity to change the limit on their nodes.

File: nixos/modules/services/web-apps/mastodon.nix (hunk 1 above)
Comment (SuperSandro2000) [reply nixos/modules/services/web-apps/mastodon.nix:223]: Don't explain this to me but instead in the option.

Comment (Izorkin) [reply nixos/modules/services/web-apps/mastodon.nix:223]: This variant?
```
        type = lib.types.int;
        default = 500;
```

**Comment by erictapen** [comment]
@Izorkin I think you want to create a PR at tootsuite/mastodon then. But I doubt it will be merged, see https://github.com/tootsuite/mastodon/pull/5697#issuecomment-345078582 for example.

Another option would be to create an Mastodon article in the [NixOS wiki](https://nixos.wiki) and document your solution there?

**Pull request closed without merging**